hdf5-sys/src/h5s.rs

```diff
@@ -112,7 +115,7 @@ extern "C" {
     ) -> htri_t;
 }
 
-#[cfg(any(hdf5_1_12_0, hdf5_1_10_7))]
```

Comment:
@mulimoen Btw, what's this?... Shouldn't this be equivalent to just `1.10.7`?

Reply:
Yes, that is equivalent. The function was backported so I kept `hdf5_1_12_0` for simplicity. It is superfluous now